Collector/ProfilePlugin.php

```diff
@@ -58,7 +58,9 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
     {
         $profile = new Profile($this->pluginName, $this->formatter->formatRequest($request));
 
-        $this->collector->getCurrentStack()->addProfile($profile);
+        if (false !== $stack = $this->collector->getCurrentStack()) {
```

Comment:
In addition to @dbu's https://github.com/php-http/HttplugBundle/pull/134#pullrequestreview-23207240: as `getCurrentStack()` either returns a `Stack` or not, would 

```php
if ($stack = $this->collector->getCurrentStack()) {
```

perhaps suffice?